Marlin/src/module/fancheck.cpp

```diff
@@ -144,31 +145,38 @@ void FanCheck::compute_speed(int elapsedTime) {
   // Drop the error when all fans are ok
   if (!fan_error_msk && error == TachoError::REPORTED) error = TachoError::FIXED;
 
-  if (error == TachoError::FIXED && !printJobOngoing()) {
+  if (error == TachoError::FIXED && !printJobOngoing() && !printingIsPaused()) {
     error = TachoError::NONE; // if the issue has been fixed while the printer is idle, reenable immediately
     ui.reset_alert_level();
   }
 
-  LOOP_L_N(f, TACHO_COUNT) if (TEST(fan_error_msk, f) && report_speed_error(f)) errors_count[f] = 0;
+  if (!enabled) return;
+
+  if (fan_error_msk & ~fan_reported_errors_msk) {
+    // Handle new faults only
+    LOOP_L_N(f, TACHO_COUNT) if (TEST(fan_error_msk, f)) report_speed_error(f);
+  }
+  fan_reported_errors_msk = fan_error_msk;
 }
 
-bool FanCheck::report_speed_error(uint8_t fan) {
+void FanCheck::report_speed_error(uint8_t fan) {
   if (printJobOngoing()) {
-    if (error != TachoError::NONE) return false;  // Keep error pending (another fan fault message is already signaled)
-    if (thermalManager.degTargetHotend(fan) != 0) {
-      TERN_(HAS_DISPLAY, ui.abort_print());
+    if (error == TachoError::NONE) {
+      if (thermalManager.degTargetHotend(fan) != 0) {
+        TERN(HAS_DISPLAY, ui.abort_print(), kill(GET_TEXT_F(MSG_FAN_SPEED_FAULT)));
```

Comment:
Is there any reason why `abort_print` doesn't abort?
repetier gets '/action:cancel' but this seems doesn't stop. Any other way to stop?
it also seems that ui.pause_print() doesn't pause at all. Maybe M25 doesn't works with usb? What I see is printer does something and starts again.

Reply:
> Maybe M25 doesn't works with usb?

That is correct. USB printing is expected to be paused from the host side. There is no G-code to pause a host print since the host can simply stop sending commands. For the benefit of host printing the `ui.pause_print` method sends a message to the host to pause itself, but only if `HOST_PROMPT_SUPPORT` is enabled. The host will only pause itself if it supports host prompts.

Reply:
I don't like to use kill but it seems the most reliable thing to do...

Reply:
It's appropriate to use `kill` when there is a safety issue, and a broken fan would count.